packages/realm/src/Dictionary.ts

```diff
@@ -217,4 +220,18 @@ export class Dictionary<T = unknown> extends Collection<string, T, [string, T],
       throw new Error(`Failed to remove missing key${keySuffix} from dictionary: ${keysSummary}`);
     }
   }
+
+  // /**
+  //  * Returns a plain object representation with possible circular references
+  //  * from the dictionary for JSON serialization.
+  //  * @returns A plain object
+  //  */
+  // @ts-expect-error We're exposing methods in the users value namespace
+  toJSON(_?: string, cache = new JSONCacheMap()): DefaultObject {
+    return Object.fromEntries(
+      Object.entries(this)
+        .filter(([k]) => k != "toJSON")
```

Comment:
I'm surprised this would ever happen, perhaps you're missing a non-enumerable property definition of the `toJSON` in the dictionary?

Reply:
I will need to check this again, it may actually not be happening since I originally wrote this when I was adding toJSON to dictionaries in a different way

Reply:
yeah this is not necessary. I removed it.